pkg/kubectl/resource/visitor.go

```diff
@@ -224,17 +225,60 @@ type URLVisitor struct {
 }
 
 func (v *URLVisitor) Visit(fn VisitorFunc) error {
-	res, err := http.Get(v.URL.String())
+	body, err := v.readHttpWithRetries(httpgetImpl, time.Second, v.URL.String())
 	if err != nil {
 		return err
 	}
-	defer res.Body.Close()
-	if res.StatusCode != 200 {
-		return fmt.Errorf("unable to read URL %q, server reported %d %s", v.URL, res.StatusCode, res.Status)
+	defer body.Close()
+	v.StreamVisitor.Reader = body
+	return v.StreamVisitor.Visit(fn)
+}
+
+// readHttpWithRetries tries to http.Get the v.URL 3 times before giving up.
+func (v *URLVisitor) readHttpWithRetries(get httpget, duration time.Duration, u string) (io.ReadCloser, error) {
+	var err error
+	var body io.ReadCloser
+	for i := 0; i < 3; i++ {
```

Comment:
Yes I think that makes sense.  Sorry for the inattention, been trying to
sort out the proto issues this morning.

On Mon, May 9, 2016 at 8:12 PM, Phillip Wittrock <notifications@github.com>
wrote:

> In pkg/kubectl/resource/visitor.go
> <https://github.com/kubernetes/kubernetes/pull/25267#discussion_r62594767>
> :
>
> >  	if err != nil {
> >  		return err
> >  	}
> > -	defer res.Body.Close()
> > -	if res.StatusCode != 200 {
> > -		return fmt.Errorf("unable to read URL %q, server reported %d %s", v.URL, res.StatusCode, res.Status)
> > +	defer body.Close()
> > +	v.StreamVisitor.Reader = body
> > +	return v.StreamVisitor.Visit(fn)
> > +}
> > +
> > +// readHttpWithRetries tries to http.Get the v.URL 3 times before giving up.
> > +func (v *URLVisitor) readHttpWithRetries(get httpget, duration time.Duration, u string) (io.ReadCloser, error) {
> > +	var err error
> > +	var body io.ReadCloser
> > +	for i := 0; i < 3; i++ {
>
> SGTM. WDYT about giving the same treatment to sleep duration?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/25267/files/abc24e04bfd8696964949aa60753c285e303b867#r62594767>
>